component/prometheus/fanout.go

```diff
@@ -45,6 +46,9 @@ func (f *Fanout) Appender(ctx context.Context) storage.Appender {
 	f.mut.RLock()
 	defer f.mut.RUnlock()
 
+	ctx = scrape.ContextWithMetricMetadataStore(ctx, NoopMetadataStore{})
+	ctx = scrape.ContextWithTarget(ctx, &scrape.Target{})
```

Comment:
Can we add a note explaining why we need this for now and a TODO to remove it when possible?